### PR TITLE
Initialize `SplitExplicitFreeSurface` with a CFL condition

### PR DESCRIPTION
**Pull request by simone-silvestri**
This PR allows initializing the `SplitExplicitFreeSurface` with the following keyword arguments:

```julia
free_surface = SplitExplicitFreeSurface(; CFL = 0.7, Δt_max = 10minutes, grid)
```

### CODE REVIEW

**Comment by glwagner** [comment]
`TimeStepWizard` uses `cfl` rather than `CFL`. So either we should change the wizard or use `cfl` here

File: src/Models/HydrostaticFreeSurfaceModels/split_explicit_free_surface.jl [hunk 1]
```diff
@@ -247,9 +247,32 @@ Return `SplitExplicitSettings`. For a description of the keyword arguments, see
 the [`SplitExplicitFreeSurface`](@ref).
 """
 function SplitExplicitSettings(FT::DataType=Float64;
-                               substeps = 200, 
+                               substeps = nothing, 
+                               CFL    = nothing,
```
Comment (glwagner) [inline src/Models/HydrostaticFreeSurfaceModels/split_explicit_free_surface.jl:251]:
```suggestion
                               cfl    = nothing,
```

File: src/Models/HydrostaticFreeSurfaceModels/split_explicit_free_surface.jl [hunk 2]
```diff
@@ -247,9 +247,32 @@ Return `SplitExplicitSettings`. For a description of the keyword arguments, see
...
+                               grid   = nothing,
+                               Δt_max = nothing,
```
Comment (glwagner) [inline src/Models/HydrostaticFreeSurfaceModels/split_explicit_free_surface.jl:253]:
```suggestion
                               max_Δt = nothing,
```

**Review by navidcy — APPROVED** [review]
Would it be useful to inform the user regarding how many substeps their cfl choice implied?

**Comment by simone-silvestri** [comment]
`show(io::IO, free_surface::SplitExplicitFreeSurface)` shows the number of substeps, 
so that information can be retrieved with `@show free_surface` (or `@show model`)

**Pull request merged**